src/python/pants/backend/python/util_rules/local_dists_pep660.py

```diff
@@ -0,0 +1,539 @@
+Build: 0.editable
+'''.format(tags))
+
+if direct_url:
+    # We abuse pex to get PEP660 editable wheels installed in a virtualenv.
+    # Pex and pip do not know that this is an editable install.
+    # We can't add direct_url.json to the wheel, because that has to be added
+    # by the wheel installer. So, we will rename this file once installed.
+    direct_url_file = os.path.join(build_dir, metadata_path, "direct_url__pants__.json")
+    with open(direct_url_file , "w") as f:
+        f.write('''\
+{{{{
+    "url": "{{}}",
+    "dir_info": {{{{
+        "editable": true
+    }}}}
+}}}}
+'''.format(direct_url))
+
+pth_file_arcname = pkg + "__pants__.pth"
+
+_record = []
+def record(file_path, file_arcname):
+    with open(file_path, "rb") as f:
+        file_digest = hashlib.sha256(f.read()).digest()
+    file_hash = "sha256=" + base64.urlsafe_b64encode(file_digest).decode().rstrip("=")
+    file_size = str(os.stat(file_path).st_size)
+    _record.append(",".join([file_arcname, file_hash, file_size]))
+
+wheel_path = "{{}}-{{}}-0.editable-{{}}.whl".format(pkg, version, tags)
+with zipfile.ZipFile(os.path.join(dist_dir, wheel_path), "w") as whl:
+    record(pth_file_path, pth_file_arcname)
+    whl.write(pth_file_path, pth_file_arcname)
+    # based on wheel.wheelfile.WheelFile.write_files (by @argonholm MIT license)
+    for root, dirnames, filenames in os.walk(os.path.join(build_dir, metadata_path)):
+        dirnames.sort()
+        for name in sorted(filenames):
+            path = os.path.normpath(os.path.join(root, name))
+            if os.path.isfile(path):
+                arcname = os.path.relpath(path, build_dir).replace(os.path.sep, "/")
+                record(path, arcname)
+                whl.write(path, arcname)
+    record_path = os.path.join(metadata_path, "RECORD")
+    _record.extend([record_path + ",,", ""])  # "" to add newline at eof
+    whl.writestr(record_path, os.linesep.join(_record))
+
+print("editable_path: {{editable_path}}".format(editable_path=wheel_path))
+"""
+
+
+def interpolate_backend_wrapper(  # based on interpolate_backend_shim
+    dist_dir: str,
+    pth_file_path: str,
+    direct_url: str,
+    request: DistBuildRequest,
+) -> bytes:
+    # See https://www.python.org/dev/peps/pep-0517/#source-trees.
+    module_path, _, object_path = request.build_system.build_backend.partition(":")
+    backend_object = f"{module_path}.{object_path}" if object_path else module_path
+
+    def config_settings_repr(cs: FrozenDict[str, tuple[str, ...]] | None) -> str:
+        # setuptools.build_meta expects list values and chokes on tuples.
+        # We assume/hope that other backends accept lists as well.
+        return distutils_repr(None if cs is None else {k: list(v) for k, v in cs.items()})
+
+    # tag the editable wheel as widely compatible
+    lang_tag, abi_tag, platform_tag = "py3", "none", "any"
+    if request.interpreter_constraints.includes_python2():
+        # assume everything has py3 support.
```

Comment:
I've been corrected in the past that comments should be full valid sentences, and thus begin with a capital letter and ending with a period  :)

Reply:
That was probably me, being a grammar nut... 

But yeah, when I see complete sentences it reassures me that the author completed their point.

Reply:
It was @Eric-Arellano actually, but yea ;)